List testing

energyc0 opened the pull request:
Added sorted list testing.

energyc0 committed:
Added recursive list library and recursive list exercise.

energyc0 committed:
CMakeLists support

energyc0 committed:
Added additional translation unit for testing sorted list.
Now CMakeLists can be configured in Debug and Release modes.
Sorted list testing is included only in Debug mode.

energyc0 committed:
Formatted code.

energyc0 committed:
Added comments

energyc0 commented:
Resolved conflicts with 'main' branch.

energyc0 committed:
Resolved conflicts with 'main' branch, added NULL pointer handling.
All sorted list functions return false on NULL pointer
Use ADT in sortedList_exercise.c

energyc0 commented:
Resolved conflicts with 'main' again.

WoWaster commented on 06.10.2025/sortedListTesting.c:
```diff
@@ -0,0 +1,127 @@
+
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof(arr[0]))
+
+#define testFprintf(stream, ...)                  \
```
Давайте всё-таки называть define в `ALL_CAPS`, иначе потом непонятно, что функция, а что макрос. Да и тут как будто бы это всё можно нормально в функции вынести

WoWaster commented on 06.10.2025/sortedListTesting.c:
```diff
@@ -0,0 +1,127 @@
+        testSortedListExpectIndex(pList, i, sortedValues[i], testResult);
+
+    if (sortedListRemoveIndex(pList, -1))
+        testErrPrintln("sortedListRemoveIndex() - success", testResult);
```
Тут точно `success`? Скорее failed если вдруг вернулось `true`

energyc0 committed:
Refactored code.
Changed macro-functions to functions int sortedListTesting.c
Changed error messages.
